### PR TITLE
fix: user page get data error

### PR DESCRIPTION
**Pull request by kittybest**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- fix user page bugs --> it's because we actually pass hex number to backend, causing action data not be found
- update user page UI/UX 
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by kittybest** [comment]
have to wait for Unirep-Social contract being modified to record spent from rep or subsidy.

**Comment by kittybest** [comment]
Actually, because we cannot tell if the spent is from reputation or subsidy, the records now looks awkward, but I think it will be easy to modify in the future as long as we could distinguish these two kind of spent. 
Now the page will not crashed when the user click anywhere, so I think this is better to be merged into main before the DevCon, even though the records now not showing correct information.

File: .gitignore [hunk 1]
```diff
@@ -17,4 +17,15 @@ packages/frontend/cypress/screenshots
 
 .node-version
 packages/core/contracts/*Verifier.sol
+
+# sqlite
+*.sql
+*/*.sql
+*/*/*.sql
+*/*/*/*.sql
+*/*/*/*/*.sql
```
Comment (vivianjeng) [inline .gitignore:26]: you can use `**/**.sql` and `**/**.sqlite`

**Comment by kittybest** [comment]
Now the activity records are still linked to wrong post/comment, and it's related to `core/synchronizer.ts` and the backend routers, I will fix them in the other PR.

File: .gitignore (hunk 1 above)
Comment (vivianjeng) [reply .gitignore]: sorry I think it should be `**/*.sql` and `**/*.sqlite`

**Review by vivianjeng — APPROVED** [review]
LGTM

File: packages/frontend/src/pages/userPage/userPage.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ const UserPage = () => {
     const [sort, setSort] = useState<QueryType>(QueryType.Boost)
     const [isDropdown, setIsDropdown] = useState<boolean>(false)
 
-    const [received, setReceived] = useState<number[]>([0, 0, 0]) // airdrop, boost, squash
+    const [received, setReceived] = useState<number[]>([0, 0]) // boost, squash
```
[the remark this answers is not in the document]
Comment (kittybest) [reply packages/frontend/src/pages/userPage/userPage.tsx:73]: do we have to modify this in this PR?

File: packages/frontend/src/context/Post.ts [hunk 3]
```diff
@@ -88,10 +88,11 @@ export class Data {
     }
 
     async loadFeed(query: string, lastRead = '0', epks = [] as string[]) {
+        const epksBase10 = epks.map((epk) => Number('0x' + epk))
```
Comment (vimwitch) [inline packages/frontend/src/context/Post.ts:91]: It's not safe to use a `Number` here if we have epoch keys larger than 64 bits. Maybe use `BigInt` instead?

**Pull request merged**